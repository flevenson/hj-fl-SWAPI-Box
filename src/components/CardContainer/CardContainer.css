.card-container {
  display: grid;
  grid-gap: 5px;
  grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
  grid-template-rows: 1fr 1fr;
  padding-top: 2rem;
}