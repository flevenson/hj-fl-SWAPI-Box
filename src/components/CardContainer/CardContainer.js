import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';
import './CardContainer.css';

const CardContainer = ({ people, planets, vehicles, display, selected, favorites, addToFavorites, favorited, isLoading }) => {
	
  // let cards

  const cards = Object.keys(display).map(element => { 
    console.log(`${selected}`)
		return <Card
			data={display[element]}
			key={element}
      id={display[element].name}
      selected={selected}
      addToFavorites={addToFavorites}
      favorited={false}
		/>
	})

  // switch(selected) {
  //   case 'people': 
  //     cards = Object.keys(people).map(element => { 
  //       console.log(`${selected}`)
  //       return <Card
  //         data={people[element]}
  //         key={element}
  //         id={people[element].name}
  //         selected={selected}
  //         addToFavorites={addToFavorites}
  //         favorited={false}
  //       />
  //     })
  //     return cards;
  //   case 'planets': 
  //     cards = Object.keys(planets).map(element => { 
  //       console.log(`${selected}`)
  //       return <Card
  //         data={planets[element]}
  //         key={element}
  //         id={planets[element].name}
  //         selected={selected}
  //         addToFavorites={addToFavorites}
  //         favorited={false}
  //       />
  //     })
  //     return cards;
  //   case 'vehicles': 
  //     cards = Object.keys(vehicles).map(element => { 
  //       console.log(`${selected}`)
  //       return <Card
  //         data={vehicles[element]}
  //         key={element}
  //         id={vehicles[element].name}
  //         selected={selected}
  //         addToFavorites={addToFavorites}
  //         favorited={false}
  //       />
  //     })
  //     return cards;
  // }

  if (!selected.length && !Object.keys(display).length) {
    return (
      <div>
        Welcome To SWAPI BOX
      </div>
  )} else if (isLoading) {
      return (
        <div>loading, please wait..</div>
    )
  }	else {
  	return (
  	  <div className='card-container'>
  	  	{ cards }
  	  </div>
  	)
  }
}

CardContainer.propTypes = {
	display: PropTypes.object.isRequired,
  selected: PropTypes.string.isRequired
}; 

export default CardContainer;