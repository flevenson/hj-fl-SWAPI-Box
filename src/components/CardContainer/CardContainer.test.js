import React from 'react';
import { shallow } from 'enzyme';

import CardContainer from './CardContainer';

describe('CardContainer', () => {

	it('should match a snapshot', () => {
		const mockData = {
			name: 'Joe',
			gender: 'male',
			height: 165 
		}
		const wrapper = shallow(<CardContainer 
									people={mockData} 
									display={{}} 
									selected={''}
								/>);
		expect(wrapper).toMatchSnapshot()
	})
})