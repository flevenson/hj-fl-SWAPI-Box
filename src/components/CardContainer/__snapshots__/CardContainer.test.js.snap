// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match a snapshot after display has been populated 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <CardContainer
    display={
      Object {
        "0": Array [
          Object {
            "homeworld": "earth",
            "name": "Joe",
            "population": 100,
            "species": "human",
          },
        ],
        "2": Array [
          Object {
            "homeworld": "Tattooine",
            "name": "Luke",
            "population": 75,
            "species": "human",
          },
        ],
      }
    }
    selected="people"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "display": Object {
        "0": Array [
          Object {
            "homeworld": "earth",
            "name": "Joe",
            "population": 100,
            "species": "human",
          },
        ],
        "2": Array [
          Object {
            "homeworld": "Tattooine",
            "name": "Luke",
            "population": 75,
            "species": "human",
          },
        ],
      },
      "selected": "people",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="card-container"
      >
        <div
          class="card"
        >
          <h1 />
          <p>
            Homeworld: 
          </p>
          <p>
            Species: 
          </p>
          <p>
            Population: 
          </p>
          <img
            alt="favorite button"
            class="favorite-button"
            src="star.svg"
          />
        </div>
        <div
          class="card"
        >
          <h1 />
          <p>
            Homeworld: 
          </p>
          <p>
            Species: 
          </p>
          <p>
            Population: 
          </p>
          <img
            alt="favorite button"
            class="favorite-button"
            src="star.svg"
          />
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
              Array [
                Object {
                  "homeworld": "earth",
                  "name": "Joe",
                  "population": 100,
                  "species": "human",
                },
              ]
            }
            selected="people"
          />,
          <Card
            data={
              Array [
                Object {
                  "homeworld": "Tattooine",
                  "name": "Luke",
                  "population": 75,
                  "species": "human",
                },
              ]
            }
            selected="people"
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "homeworld": "earth",
                "name": "Joe",
                "population": 100,
                "species": "human",
              },
            ],
            "selected": "people",
          },
          "ref": null,
          "rendered": Object {
            "instance": <div
              class="card"
            >
              <h1 />
              <p>
                Homeworld: 
              </p>
              <p>
                Species: 
              </p>
              <p>
                Population: 
              </p>
              <img
                alt="favorite button"
                class="favorite-button"
                src="star.svg"
              />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <p>
                  Homeworld: 
                </p>,
                <p>
                  Species: 
                </p>,
                <p>
                  Population: 
                </p>,
                <img
                  alt="favorite button"
                  className="favorite-button"
                  src="star.svg"
                />,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h1 />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "h1",
              },
              Object {
                "instance": <p>
                  Homeworld: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Homeworld: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Homeworld: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  Species: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Species: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Species: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  Population: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Population: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Population: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <img
                  alt="favorite button"
                  class="favorite-button"
                  src="star.svg"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "favorite button",
                  "className": "favorite-button",
                  "src": "star.svg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "homeworld": "Tattooine",
                "name": "Luke",
                "population": 75,
                "species": "human",
              },
            ],
            "selected": "people",
          },
          "ref": null,
          "rendered": Object {
            "instance": <div
              class="card"
            >
              <h1 />
              <p>
                Homeworld: 
              </p>
              <p>
                Species: 
              </p>
              <p>
                Population: 
              </p>
              <img
                alt="favorite button"
                class="favorite-button"
                src="star.svg"
              />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <p>
                  Homeworld: 
                </p>,
                <p>
                  Species: 
                </p>,
                <p>
                  Population: 
                </p>,
                <img
                  alt="favorite button"
                  className="favorite-button"
                  src="star.svg"
                />,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h1 />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "h1",
              },
              Object {
                "instance": <p>
                  Homeworld: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Homeworld: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Homeworld: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  Species: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Species: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Species: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  Population: 
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Population: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Population: ",
                ],
                "type": "p",
              },
              Object {
                "instance": <img
                  alt="favorite button"
                  class="favorite-button"
                  src="star.svg"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "favorite button",
                  "className": "favorite-button",
                  "src": "star.svg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "display": Object {
          "0": Array [
            Object {
              "homeworld": "earth",
              "name": "Joe",
              "population": 100,
              "species": "human",
            },
          ],
          "2": Array [
            Object {
              "homeworld": "Tattooine",
              "name": "Luke",
              "population": 75,
              "species": "human",
            },
          ],
        },
        "selected": "people",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="card-container"
        >
          <div
            class="card"
          >
            <h1 />
            <p>
              Homeworld: 
            </p>
            <p>
              Species: 
            </p>
            <p>
              Population: 
            </p>
            <img
              alt="favorite button"
              class="favorite-button"
              src="star.svg"
            />
          </div>
          <div
            class="card"
          >
            <h1 />
            <p>
              Homeworld: 
            </p>
            <p>
              Species: 
            </p>
            <p>
              Population: 
            </p>
            <img
              alt="favorite button"
              class="favorite-button"
              src="star.svg"
            />
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              data={
                Array [
                  Object {
                    "homeworld": "earth",
                    "name": "Joe",
                    "population": 100,
                    "species": "human",
                  },
                ]
              }
              selected="people"
            />,
            <Card
              data={
                Array [
                  Object {
                    "homeworld": "Tattooine",
                    "name": "Luke",
                    "population": 75,
                    "species": "human",
                  },
                ]
              }
              selected="people"
            />,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "homeworld": "earth",
                  "name": "Joe",
                  "population": 100,
                  "species": "human",
                },
              ],
              "selected": "people",
            },
            "ref": null,
            "rendered": Object {
              "instance": <div
                class="card"
              >
                <h1 />
                <p>
                  Homeworld: 
                </p>
                <p>
                  Species: 
                </p>
                <p>
                  Population: 
                </p>
                <img
                  alt="favorite button"
                  class="favorite-button"
                  src="star.svg"
                />
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <p>
                    Homeworld: 
                  </p>,
                  <p>
                    Species: 
                  </p>,
                  <p>
                    Population: 
                  </p>,
                  <img
                    alt="favorite button"
                    className="favorite-button"
                    src="star.svg"
                  />,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <h1 />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": <p>
                    Homeworld: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Homeworld: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Homeworld: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p>
                    Species: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Species: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Species: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p>
                    Population: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Population: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Population: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <img
                    alt="favorite button"
                    class="favorite-button"
                    src="star.svg"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "favorite button",
                    "className": "favorite-button",
                    "src": "star.svg",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "img",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "homeworld": "Tattooine",
                  "name": "Luke",
                  "population": 75,
                  "species": "human",
                },
              ],
              "selected": "people",
            },
            "ref": null,
            "rendered": Object {
              "instance": <div
                class="card"
              >
                <h1 />
                <p>
                  Homeworld: 
                </p>
                <p>
                  Species: 
                </p>
                <p>
                  Population: 
                </p>
                <img
                  alt="favorite button"
                  class="favorite-button"
                  src="star.svg"
                />
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <p>
                    Homeworld: 
                  </p>,
                  <p>
                    Species: 
                  </p>,
                  <p>
                    Population: 
                  </p>,
                  <img
                    alt="favorite button"
                    className="favorite-button"
                    src="star.svg"
                  />,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <h1 />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": <p>
                    Homeworld: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Homeworld: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Homeworld: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p>
                    Species: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Species: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Species: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p>
                    Population: 
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Population: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Population: ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <img
                    alt="favorite button"
                    class="favorite-button"
                    src="star.svg"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "favorite button",
                    "className": "favorite-button",
                    "src": "star.svg",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "img",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should match a snapshot when display is empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    display={Object {}}
    selected=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Welcome To SWAPI BOX",
    },
    "ref": null,
    "rendered": "Welcome To SWAPI BOX",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Welcome To SWAPI BOX",
      },
      "ref": null,
      "rendered": "Welcome To SWAPI BOX",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
